src/app/components/lineage-graph/lineage-graph.component.ts

```diff
@@ -23,10 +20,10 @@ import { LineageGraphNodeKind } from "src/app/dataset-view/additional-components
     styleUrls: ["./lineage-graph.component.scss"],
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class LineageGraphComponent implements OnChanges, OnInit {
-    @Input() public links: Edge[];
-    @Input() public nodes: Node[];
-    @Input() public currentDataset: DatasetLineageBasicsFragment;
+export class LineageGraphComponent {
+    @Input() public links: MaybeNull<Edge[]>;
+    @Input() public nodes: MaybeNull<Node[]>;
```

Comment:
I think the entire graph needs an explicit model.
It's hard to imagine that edges arrived, and nodes didn't. It's either both or none.
So, I think we should have:
   `@Input() public graph: MaybeNull<LineageGraph>;`
   
 where 
 
 ```
 interface LineageGraph {
   nodes: Node[];
   edges: Edge[];
 }
 ```

Reply:
Fixed